chained/water_impl_test.go

```diff
@@ -0,0 +1,100 @@
+package chained
+
+import (
+	"context"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/getlantern/common/config"
+	"github.com/getlantern/flashlight/v7/ops"
+	"github.com/refraction-networking/water"
+	_ "github.com/refraction-networking/water/transport/v0"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewWaterImpl(t *testing.T) {
+	type params struct {
+		raddr          string
+		pc             *config.ProxyConfig
+		reportDialCore reportDialCoreFn
+	}
+
+	f, err := os.ReadFile("../wasm/reverse.go.wasm")
+	require.Nil(t, err)
+
+	var tests = []struct {
+		name        string
+		givenParams params
+		assert      func(t *testing.T, actual *waterImpl, err error)
+	}{
+		{
+			name: "create new waterImpl with success",
+			givenParams: params{
+				raddr: "127.0.0.1",
+				pc: &config.ProxyConfig{
+					PluggableTransportSettings: map[string]string{
+						"water_wasm": string(f),
+					},
+				},
+				reportDialCore: func(op *ops.Op, dialCore func() (net.Conn, error)) (net.Conn, error) {
+					return nil, nil
+				},
+			},
+			assert: func(t *testing.T, actual *waterImpl, err error) {
+				assert.Nil(t, err)
+				assert.NotNil(t, actual)
+				assert.NotNil(t, actual.config)
```

Comment:
Thanks for the suggestions! I've just implemented this a little bit different (defined `wantConfig` before tests  because I didn't want to pass as an assert argument since I'm just receiving the parameters returned from the function in scope).